src/ray/raylet/scheduling/cluster_resource_scheduler.cc

```diff
@@ -1000,17 +1002,28 @@ void ClusterResourceScheduler::FillResourceUsage(
       (*resources_data->mutable_resources_total())[label] = capacity.total.Double();
     }
   }
-  for (const auto &it : resources.custom_resources) {
+  for (auto &it : resources.custom_resources) {
     uint64_t custom_id = it.first;
-    const auto &capacity = it.second;
+    auto &capacity = it.second;
     const auto &last_capacity = last_report_resources_->custom_resources[custom_id];
     const auto &label = string_to_int_map_.Get(custom_id);
+
+    // Automatically report object store usage.
+    // XXX: this MUTATES the resources field, which is needed since we are storing
+    // it in last_report_resources_.
+    if (label == "object_store_memory" && get_used_object_store_memory_ != nullptr) {
```

Comment:
I don't actually understand how any of the resource struct works but should this be a "predefined resource" instead of a custom one?

Then, we can also modify outside of the for loops.

Reply:
Let me replace TPU with this...